### PR TITLE
Added some docs for AggregatedNode

### PR DESCRIPTION
**Pull request by snorfalorpagus**
In response to #754.

### CODE REVIEW

File: docs/source/cookbook/aggregated_node.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
+Aggregated nodes
+----------------
+
+The `AggregatedNode` is a special node that allows constraints to be added to a model that apply to a group of (possibly disconnected) nodes, enforcing either a fixed ratio or fixed maximum.
```
Comment (jetuk) [inline docs/source/cookbook/aggregated_node.rst]: `.... enforcing either a fixed ratio of flow and/or a total maximum flow.`

Is this more accurate? I assume the solver supports minimum flow too?

File: docs/source/cookbook/aggregated_node.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+Constraining flow using a ratio
+===============================
+
+Aggregated nodes allows a constraint to be added that ensures the flow via two or more nodes conforms to a specific ratio. This is useful, for example, when modelling a blending constraint between multiple sources of water.
+
+In the example below the aggregated node "D" constrains the flow in nodes "A" and "B" to be equal (0.5 + 0.5 == 1). Due to the constraint the solution is flow from A = 40, B = 40, despite C demanding 100.
```
Comment (jetuk) [inline docs/source/cookbook/aggregated_node.rst]: The `0.5 + 0.5 == 1` is not a requirement? It simply uses relative weights? One could also do this with factors of 50 and 50?

File: docs/source/cookbook/aggregated_node.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        {
+            "name": "B",
+            "type": "input",
+            "max_flow": 40.0
+        },
+        {
+            "name": "D",
+            "type": "aggregated",
+            "nodes": ["A", "B"],
+            "max_flow": 60.0
```
Comment (jetuk) [inline docs/source/cookbook/aggregated_node.rst]: I guess we shoud also show an example that uses the "flow_weights" key.

Comment (jetuk) [reply docs/source/cookbook/aggregated_node.rst]: This can also be a parameter.

File: docs/source/cookbook/aggregated_node.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            "name": "C",
+            "type": "output",
+            "max_flow": 100.0,
+            "cost": -10.0
+        },
+        {
+            "name": "D",
+            "type": "aggregated",
+            "nodes": ["A", "B"],
+            "factors": [0.5, 0.5]
```
Comment (jetuk) [inline docs/source/cookbook/aggregated_node.rst]: We should comment that these ratios must (currently) be constant for an entire simulation.

**Comment by Batch21** [comment]
I was just playing around with aggregated nodes and noticed that if you try and add two nodes from the same route to an aggregated node it crashes Python with this c error:

```
glp_set_mat_row: i = 8; ind[2] = 1; duplicate column indices not allowed
Error detected in file ..\src\api\prob1.c at line 779
```

If you try and do this within a notebook, there is no indication about what went wrong. The kernel just cashes.

I know you probably shouldn't be doing this (or need to) but it might be worth putting a explicit warning in the doc string about it.

**Pull request merged**